C/Coin-OR/Bonmin/build_tarballs.jl

```diff
@@ -0,0 +1,71 @@
+using BinaryBuilder, Pkg
+
+name = "Bonmin"
+version = v"1.8.8"
+
+sources = [
+    GitSource("https://github.com/coin-or/Bonmin.git",
+              "65c56cea1e7c40acd9897a2667c11f91d845bb7b"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/Bonmin
+
+if [[ ${target} == *mingw* ]]; then
+    sed -i s/dllimport/dllexport/ /workspace/destdir/include/coin-or/IpoptConfig.h
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fix_dllexport.patch
+fi
+
+# Remove misleading libtool files
+rm -f ${libdir}/*.la
+update_configure_scripts
+
+# old and custom autoconf
+sed -i s/elf64ppc/elf64lppc/ configure
+
+export CPPFLAGS="${CPPFLAGS} -I${includedir} -I${includedir}/coin -I${includedir}/coin-or"
+export CXXFLAGS="${CXXFLAGS} -std=c++11"
+
+if [[ ${target} == *mingw* ]]; then
+    export LDFLAGS="-L${libdir}"
+    export LT_LDFLAGS="-no-undefined"
+fi
+
+./configure \
+    --prefix=${prefix} \
+    --build=${MACHTYPE} \
+    --host=${target} \
+    --disable-pthread-mumps \
+    --enable-static \
+    --enable-shared \
+    lt_cv_deplibs_check_method=pass_all \
+    --with-asl-lib="-lipoptamplinterface -lasl"
+
+make
+make install
+"""
+
+
+platforms = supported_platforms()
+filter!(!Sys.isfreebsd, platforms)
+platforms = expand_cxxstring_abis(platforms)
+filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
+platforms = expand_gfortran_versions(platforms)
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libbonmin", :libbonmin),
+    ExecutableProduct("bonmin", :amplexe),
```

Comment:
Is the fact that the product variable is different from the executable name on purpose?

Reply:
Yes. There is precedent with Ipopt. 
https://github.com/JuliaPackaging/Yggdrasil/blob/fb52818b44fc19d183819b7eacc60082f6718200/I/Ipopt/build_tarballs.jl#L43

So it would make sense if people just replace `Ipopt_jll.amplexe() do path` with `Bonmin_jll.amplexe() do path`.

The binary is really just a binary for calling `bonmin` via AMPL (ASL_jll is the AMPL Solver Library).

Reply:
Ok ok, up to you, I just wanted to be sure this wasn't an obvious error :slightly_smiling_face: 

Related, note that the new preferred (thread-safe!) syntax to call the executables is
```julia
run(`$(bonmin()) ...`)
```
see https://docs.binarybuilder.org/stable/jll/#ExecutableProduct.  But this is only available in Julia v1.6+, so you may not be able to use it.

Reply:
Nice. I'm very keen to see 1.6 become the LTS so we can drop all of our 1.0 support and use this sort of stuff. 

Thanks for all your work, and apologies for the obvious issues I stumble over.

Reply:
No worries!  And nice to see that after all the troubles here the final build had only very few warnings!